### PR TITLE
Import: Add new CPUs for PPC and honor `asm.cpu` for `asm.arch=ppc.gnu`

### PR DESCRIPTION
**Pull request by officialcjunior**
**Your checklist for this pull request**
- [x] I've read the [guidelines for contributing](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md) to this repository
- [x] I made sure to follow the project's [coding style](https://github.com/rizinorg/rizin/blob/master/DEVELOPERS.md#code-style)
- [ ] I've added tests that prove my fix is effective or that my feature works (if possible)
- [ ] I've updated the documentation and the [rizin book](https://github.com/rizinorg/book) with the relevant information (if needed)

**Detailed description**
Imports two commits from radare2:
*    https://github.com/radareorg/radare2/commit/c150dc489fe68c3e02750ef72cc365c5cc5f549a
 *   https://github.com/radareorg/radare2/commit/0ace091ff4b95c194ef975f156e31150656e7ee9

**Test plan**
CI is green.

**Closing issues**
Partially addresses https://github.com/rizinorg/rizin/issues/747

### CODE REVIEW

File: librz/asm/p/asm_ppc_gnu.c [hunk 1]
```diff
@@ -51,6 +52,14 @@ static int disassemble(RzAsm *a, RzAsmOp *op, const ut8 *buf, int len) {
 	/* prepare disassembler */
 	memset(&disasm_obj, '\0', sizeof(struct disassemble_info));
 	disasm_obj.disassembler_options = (a->bits == 64) ? "64" : "";
```
Comment (ret2libc) [inline librz/asm/p/asm_ppc_gnu.c]: this should not be necessary anymore

File: librz/asm/p/asm_ppc_gnu.c [hunk 2]
```diff
@@ -51,6 +52,14 @@ static int disassemble(RzAsm *a, RzAsmOp *op, const ut8 *buf, int len) {
...
+	*options = 0;
+	if (!RZ_STR_ISEMPTY(a->cpu)) {
+		snprintf(options, sizeof(options), "%s,%s",
+			(a->bits == 64) ? "64" : "", a->cpu);
+	} else if (a->bits == 64) {
+		rz_str_ncpy(options, "64", sizeof(options));
+	}
+	disasm_obj.disassembler_options = options;
```
Comment (ret2libc) [inline librz/asm/p/asm_ppc_gnu.c]: If I'm not missing anything, `disasm_obj.disassembler_options` seems to be a `char *`, so assigning it to `options` which is a local variable is not ok.

**Comment by XVilka** [comment]
Maybe we should have a test for this. Also summoning @wargio as PowerPC expert.

File: librz/asm/p/asm_ppc_gnu.c (hunk 2 above)
Comment (wargio) [reply librz/asm/p/asm_ppc_gnu.c]: you need to alloc it. use rz_str_newf

Comment (officialcjunior) [reply librz/asm/p/asm_ppc_gnu.c]: `disasm_obj.disassembler_options = rz_str_newf("%s", options);`

Like that?

Comment (XVilka) [reply librz/asm/p/asm_ppc_gnu.c]: Yes.

File: librz/asm/p/asm_ppc_gnu.c (hunk 1 above)
Comment (XVilka) [reply librz/asm/p/asm_ppc_gnu.c]: @officialcjunior please try to remove this, and lets see if it works.

File: librz/asm/p/asm_ppc_gnu.c [hunk 3]
```diff
@@ -50,7 +51,14 @@ static int disassemble(RzAsm *a, RzAsmOp *op, const ut8 *buf, int len) {
...
 	memset(&disasm_obj, '\0', sizeof(struct disassemble_info));
-	disasm_obj.disassembler_options = (a->bits == 64) ? "64" : "";
+	*options = 0;
+	if (!RZ_STR_ISEMPTY(a->cpu)) {
+		snprintf(options, sizeof(options), "%s,%s",
+			(a->bits == 64) ? "64" : "", a->cpu);
+	} else if (a->bits == 64) {
+		rz_str_ncpy(options, "64", sizeof(options));
+	}
+	disasm_obj.disassembler_options = rz_str_newf("%s", options);
```
Comment (wargio) [inline librz/asm/p/asm_ppc_gnu.c]: can't you just do `disasm_obj.disassembler_options = rz_str_newf("%s,%s", (a->bits == 64) ? "64" : "", a->cpu);` in that if?

Comment (officialcjunior) [reply librz/asm/p/asm_ppc_gnu.c]: I think we can. We'll just eliminate `options[64]` and directly assign `rz_str_newf` on both of the conditions, then. 
```diff
diff --git a/librz/asm/p/asm_ppc_gnu.c b/librz/asm/p/asm_ppc_gnu.c
index 213d61f12b..e5bfee22e3 100644
--- a/librz/asm/p/asm_ppc_gnu.c
+++ b/librz/asm/p/asm_ppc_gnu.c
@@ -50,7 +50,11 @@ static int disassemble(RzAsm *a, RzAsmOp *op, const ut8 *buf, int len) {

        /* prepare disassembler */
        memset(&disasm_obj, '\0', sizeof(struct disassemble_info));
-       disasm_obj.disassembler_options = (a->bits == 64) ? "64" : "";
+       if (!RZ_STR_ISEMPTY(a->cpu)) {
+               disasm_obj.disassembler_options = rz_str_newf("%s,%s", (a->bits == 64) ? "64" : "", a->cpu);
+       } else if (a->bits == 64) {
+               disasm_obj.disassembler_options = rz_str_newf("%s", 64);
+       }
        disasm_obj.buffer = bytes;
        disasm_obj.read_memory_func = &ppc_buffer_read_memory;
        disasm_obj.symbol_at_address_func = &symbol_at_address;
```

Comment (XVilka) [reply librz/asm/p/asm_ppc_gnu.c]: It is fine, just please use `RZ_STR_NOTEMPTY(a->cpu)` instead of `!RZ_STR_ISEMPTY()`

**Review by XVilka — APPROVED** [review]

**Review by wargio — APPROVED** [review]

**Pull request merged**